core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BpmBillingProjectCreator.scala

```diff
@@ -0,0 +1,57 @@
+package org.broadinstitute.dsde.rawls.billing
+
+import akka.http.scaladsl.model.StatusCodes
+import org.broadinstitute.dsde.rawls.RawlsExceptionWithErrorReport
+import org.broadinstitute.dsde.rawls.model.{CreateRawlsV2BillingProjectFullRequest, ErrorReport, UserInfo}
+
+import scala.concurrent.{ExecutionContext, Future, blocking}
+
+
+/**
+ * This class knows how to validate Rawls billing project requests and instantiate linked billing profiles in the
+ * billing profile manager service.
+ */
+class BpmBillingProjectCreator(billingRepository: BillingRepository,
+                               billingProfileManagerDAO: BillingProfileManagerDAO)
+                              (implicit val executionContext: ExecutionContext) extends BillingProjectCreator {
+
+  override def validateBillingProjectCreationRequest(createProjectRequest: CreateRawlsV2BillingProjectFullRequest,
+                                                     userInfo: UserInfo): Future[Unit] = {
+    val azureManagedAppCoordinates = createProjectRequest.billingInfo match {
+      case Left(_) => throw new NotImplementedError("Google billing accounts not supported in billing profiles")
+      case Right(coords) => coords
+    }
+
+    for {
+      apps <- blocking {
+        billingProfileManagerDAO.listManagedApps(azureManagedAppCoordinates.subscriptionId, userInfo)
+      }
+      _ = apps.find(app => app.getSubscriptionId == azureManagedAppCoordinates.subscriptionId &&
+        app.getManagedResourceGroupId == azureManagedAppCoordinates.managedResourceGroupId &&
+        app.getTenantId == azureManagedAppCoordinates.tenantId
+      ) match {
+        case None => throw new ManagedAppNotFoundException(ErrorReport(StatusCodes.Forbidden, "Managed application not found"))
+        case Some(a) => a
```

Comment:
Nit: This structure is a bit weird: Since the value is being unwrapped and returned, it looks like it actually matters. But the only part of this that actually matters, is that it's not None.
It might a bit more clear to match with _, then return Unit.